lib/gems/pending/VMwareWebService/VimService.rb

```diff
@@ -1249,16 +1253,32 @@ def marshalObj(node, obj)
     end
   end
 
+  def handle_memory_and_gc(response)
+    # At this point we don't need the internal raw XML content since we
+    #   have the parsed XML document, so we can free it for the GC.
+    response.instance_variable_set(:@http_body, nil)
+
+    xml_len = response.document.to_s.length
```

Comment:
Does `response.document` just return an internal reference, or does it actually create something?  If the latter, we may want to pass it in because it's being created on line 1277 as well.